controllers/deployment_controller.go

```diff
@@ -133,39 +133,44 @@ func (r *DeploymentReconciler) SetupWithManager(ctx context.Context, mgr ctrl.Ma
 		return err
 	}
 
-	err = c.Watch(&source.Kind{Type: &csibaremetalv1.Deployment{}}, &handler.EnqueueRequestForObject{})
+	err = c.Watch(source.Kind(mgr.GetCache(), &csibaremetalv1.Deployment{}), &handler.EnqueueRequestForObject{})
 	if err != nil {
 		return err
 	}
 
-	err = c.Watch(&source.Kind{Type: &appsv1.Deployment{}}, &handler.EnqueueRequestForOwner{
-		IsController: true,
-		OwnerType:    &csibaremetalv1.Deployment{},
-	})
+	err = c.Watch(source.Kind(mgr.GetCache(), &appsv1.Deployment{}), handler.EnqueueRequestForOwner(
```

Comment:
just a note: the controller-runtime is a big upgrade here from v11 to v16, I am expecting there is big breaking changes across the versions and some API changes.  So, these changes are some of them, right?  but please carefully read these changed API and make sure the change is not changing the functionality or behavior of our old code.

Reply:
Yes, all these changes are related to controller-runtime upgrade. I read the API changes but I don't know the product so I don't know if behavior was changed or not? How can I if I don't know the original behavior? That's why I'm asking you to review these API changes and provide feedback if the functionality might be changed or not.